python/beeai_framework/tools/mcp/mcp.py

```diff
@@ -62,6 +63,10 @@ async def _run(self, input_data: Any, options: ToolRunOptions | None, context: R
             name=self._tool.name, arguments=input_data.model_dump(exclude_none=True, exclude_unset=True)
         )
         logger.debug(f"Tool result: {result}")
+        if result.isError:
+            if result.content and isinstance(result.content[0], TextContent):
+                raise ToolError(result.content[0].text)
+            raise ToolError(JSONToolOutput(result.content).get_text_content(), context=result.structuredContent)
```

Comment:
Why not just `raise ToolError(to_json(result.content, indent=4, sort_keys=False), context=result.structuredContent)`